src/audio/graph_thread.rs

```diff
@@ -91,6 +93,12 @@ impl AudioGraphThread {
                     }
                 }
             }
+
+            // Process a render quantum if the sink can handle
+            // more data.
+            if !self.sink.has_enough_data() {
```

Comment:
There's a slight problem here; this loop will eat up CPU instead of blocking because there's nothing to block on.

Reply:
Hmm, isn't this what's supposed to happen for the rendering thread? In any case, we could use appsrc's block property here instead of the custom `has_enough_data` check.